WEB_SERVICE/WEB/_MEDIUM/GRAPHQL_CIRCULAR_REFERENCES/recommendation.md

```diff
@@ -0,0 +1,71 @@
+To mitigate the risk of circular references in GraphQL, you can follow these recommendations:
+1. **Depth Limiting**:
+Implement a middleware to check the depth of the query, and raise an error if it exceeds the limit.
+Example:
+=== "Python"
+    ```python
+    class DepthAnalysisMiddleware:
+        def resolve(self, next, root, info, **args):
+            if info.operation.selection_set:
+                depth = 0
+                for field in info.operation.selection_set.selections:
+                    depth = max(depth, self._get_depth(field))
+                if depth > 3:
+                    raise Exception('Query depth is too high')
+            return next(root, info, **args)
+    
+        def _get_depth(self, field):
+            if field.selection_set:
+                return 1 + max(self._get_depth(f) for f in field.selection_set.selections)
+            return 1
+    ```
```

Comment:
It's broken. Please double-check the other code snippets as well.
![image](https://github.com/user-attachments/assets/13501e04-7f4d-43a6-879f-e8fec9ca045e)